controllers/tempo/tempostack_controller.go

```diff
@@ -178,11 +178,29 @@ func (r *TempoStackReconciler) reconcileManifests(ctx context.Context, log logr.
 
 	}
 
+	tenantSecerts := []*manifestutils.GatewayTenantSecret{}
+	if tempo.Spec.Tenants != nil && tempo.Spec.Tenants.Mode == v1alpha1.Static {
+		if err = gateway.ValidateModes(tempo); err != nil {
```

Comment:
this seems weird. I would expect that the `ValidateModes`  can be called regardless of what `tempo.Spec.Tenants.Mode` is set to. 

Also with the current implementation the `ValidateModes` names does not reflect what the function does.